File: app/src/main/java/n7/ad2/tournaments/db/GamesRoomDatabase.java
package n7.ad2.tournaments.db;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

@Database(entities = {TournamentGame.class}, version = 2, exportSchema = false)
public abstract class GamesRoomDatabase extends RoomDatabase {

    public static GamesRoomDatabase INSTANCE;

    public static GamesRoomDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (GamesRoomDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context, GamesRoomDatabase.class, "tournaments2.db")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return INSTANCE;
    }

    public abstract GamesDao gamesDao();
}

File: feature/tournaments/src/main/kotlin/n7/ad2/tournaments/internal/db/GamesRoomDatabase.kt
package n7.ad2.tournaments.internal.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [TournamentGame::class], version = 2, exportSchema = false)
abstract class GamesRoomDatabase : RoomDatabase() {

    abstract fun gamesDao(): GamesDao

    companion object {
        @Volatile
        private var INSTANCE: GamesRoomDatabase? = null

        fun getDatabase(context: Context): GamesRoomDatabase = INSTANCE ?: synchronized(this) {
            val instance = Room.databaseBuilder(
                context.applicationContext,
                GamesRoomDatabase::class.java,
                "tournaments2.db",
            )
                .fallbackToDestructiveMigration()
                .build()
            INSTANCE = instance
            instance
        }
    }
}
